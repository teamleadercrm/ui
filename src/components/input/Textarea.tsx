import React, { forwardRef, ReactNode } from 'react';
import Box, { BoxProps } from '../box/Box';
import cx from 'classnames';

import theme from './theme.css';
import { omitBoxProps, pickBoxProps } from '../box';
import InputBase from './InputBase';
import ValidationText from '../validationText';

interface TextareaProps extends Omit<BoxProps, 'size' | 'ref'> {
  /** Sets a class name for the wrapper to give custom styles. */
  className?: string;
  /** The text to use as error message below the input. */
  error?: boolean | ReactNode;
  /** The text to use as help text below the input. */
  helpText?: string;
  /** Boolean indicating whether the input should render as inverse. */
  inverse?: boolean;
  /** The text string/element to use as success message below the input. */
  success?: boolean | ReactNode;
  /** The text to use as warning message below the input. */
  warning?: boolean | ReactNode;
}

const Textarea = forwardRef(
  ({ className, error, helpText, inverse = false, success, warning, ...others }: TextareaProps, forwardedRef) => {
    const classNames = cx(
      theme['wrapper'],
      {
        [theme['has-error']]: error,
        [theme['has-success']]: success,
        [theme['has-warning']]: warning,
      },
      className,
    );

    const boxProps = pickBoxProps(others);
    const inputProps = {
      inverse,
      ...omitBoxProps(others),
    };

    return (
      <Box className={classNames} {...boxProps}>
        <InputBase ref={forwardedRef} className={theme['textarea']} element="textarea" {...inputProps} />
        <ValidationText error={error} help={helpText} inverse={inverse} success={success} warning={warning} />
      </Box>
    );
  },
);

export default Textarea;
