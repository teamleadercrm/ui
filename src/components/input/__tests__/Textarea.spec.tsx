import React from 'react';
import { render } from '@testing-library/react';
import Textarea from '../Textarea';

describe('Component - Textarea', () => {
  it('renders', async () => {
    const { asFragment } = render(<Textarea value="foobar" placeholder="This is a placeholder" onChange={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
