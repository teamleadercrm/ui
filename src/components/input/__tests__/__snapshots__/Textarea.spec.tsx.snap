// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Textarea renders 1`] = `
<DocumentFragment>
  <div
    class="box wrapper"
    data-teamleader-ui="textarea"
  >
    <textarea
      class="input is-medium textarea"
      placeholder="This is a placeholder"
      type="text"
    >
      foobar
    </textarea>
  </div>
</DocumentFragment>
`;
